.github/workflows/static-analysis.yaml

```diff
@@ -40,12 +40,12 @@ jobs:
     name: Shellcheck
     runs-on: ${{ inputs.runsOn }}
     steps:
-      - uses: actions/checkout@v2
-      - name: Run ShellCheck
-        uses: ludeeus/action-shellcheck@master
-        with:
-          scandir: ${{ inputs.scanDir }}
-          ignore_paths: ${{ inputs.ignoreDir }}
+      - uses: actions/checkout@v3
+      - run: |
+          wget -q https://github.com/koalaman/shellcheck/releases/download/stable/shellcheck-stable.linux.x86_64.tar.xz
+          tar -xvf shellcheck-stable.linux.x86_64.tar.xz
+          shellcheck-stable/shellcheck -V; find . -name '*.sh' -print -exec sha256sum {} \;; find . -type f \( -name "*.sh" \) -print -exec shellcheck-stable/shellcheck -a -s bash -S warning -f gcc {} \;;
```

Comment:
if not now, please add a placeholder somewhere to tag a specific version for this lint, otherwise we may get a surprise again.